data/trie/patriciaMerkleTrie.go

```diff
@@ -516,6 +516,12 @@ func (tr *patriciaMerkleTrie) Snapshot() error {
 
 	if len(tr.snapshots) > maxSnapshots {
 		dbUniqueId := strconv.Itoa(tr.snapshotId - len(tr.snapshots))
+
+		err = tr.snapshots[0].Close()
+		if err != nil {
+			tr.snapshotInProgress = false
```

Comment:
You can use var err error and a defer at line 508 which set tr.snapshotInProgress to false if err != nil to avoid setting it on false for each return in this method

Reply:
Never thought of this. Changed.